File: python/008/main.py
# ProjectEuler 008
# Python / main.py
# copyright (c) 2015 Susisu
#
# Python 3.4.2
# `python3 main.py`


def main():
    f = open("./digits.txt", "r")
    digits = f.readline()[0:-2]
    l = len(digits)
    n = 13
    m = 0
    for i in range(0, l - n + 1):
        p = 1
        for j in range(0, n):
            p *= int(digits[i + j])
        if p > m:
            m = p
    print(m)

main()

File: julia/008/main.jl
# ProjectEuler 008
# Julia / main.jl
# copyright (c) 2015 Susisu
#
# Julia v0.4.1
# `julia main.jl`

function main()
    stream = open("./digits.txt", "r")
    digits = ""
    n = 1000
    try
        digits = readline(stream)
    catch err
        warn("couldn't open file")
        return
    finally
        close(stream)
    end

    if length(digits) < n
        warn("insufficient input")
        return
    end

    max = 0
    for i = 1:(n - 12)
        p = 1
        for j = i:(i + 12)
            p *= Int(digits[j]) - Int('0')
        end
        if p > max
            max = p
        end
    end
    println(max)
end

main()
